### PR TITLE
feat(node): wsctl support message querying by message id

### PR DESCRIPTION
**Pull request by saitofun**

### CODE REVIEW

File: cmd/node/apis/types.go [hunk 1]
```diff
@@ -13,3 +13,7 @@ type msgReq struct {
 	ProjectVersion string `json:"projectVersion"   binding:"required"`
 	Data           string `json:"data"             binding:"required"`
 }
+
+type HandleRsp struct {
```
Comment (huangzhiran) [inline cmd/node/apis/types.go:17]: rename to msgResp ? for alignment with msgReq

File: cmd/wsctl/cmd/message_query.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+		switch sc := rsp.StatusCode; sc {
+		case http.StatusNotFound:
+			return errors.Errorf("the message [%s] is not found or expired", messageID)
+		case http.StatusOK:
+		default:
+			return errors.Errorf("responded status code: %d", sc)
+		}
+
+		defer rsp.Body.Close()
```
Comment (huangzhiran) [inline cmd/wsctl/cmd/message_query.go:43]: move to 34 line ?

File: vm/type.go [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+import "github.com/machinefi/w3bstream-mainnet/enums"
+
 type Type string
 
 const (
 	Risc0 Type = "risc0"
 	Halo2 Type = "halo2"
 )
+
+var vmEndpointConfigEnvKeyMap = map[string]Type{
```
Comment (huangzhiran) [inline vm/type.go:12]: no use in this pr ?

File: README.md [hunk 4]
```diff
@@ -1,77 +1,102 @@
...
+W3bstream-node need write proof to chain and need a private key for chain write. Set your private key
+at https://github.com/machinefi/sprout/blob/main/docker-compose.yaml#L20  
+If you need risc0 snark proof, a bonsai key is needed
+at https://github.com/machinefi/sprout/blob/main/docker-compose.yaml#L40
 
 ### Use customized project code
-Docker-compose will mount current work directory to containers /data https://github.com/machinefi/sprout/blob/main/docker-compose.yaml#L23  
-So you can appoint the project file at https://github.com/machinefi/sprout/blob/main/docker-compose.yaml#L18
 
+Docker-compose will mount current work directory to containers
```
Comment (huangzhiran) [inline README.md:50]: those readme  changes  for ?

File: msg/messages/message_context.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package messages
```
Comment (huangzhiran) [inline msg/messages/message_context.go:1]: maybe this package named recorder more suitable

File: Dockerfile [hunk 6]
```diff
@@ -2,7 +2,7 @@
 FROM golang:1.21 AS builder
 
 ENV GO111MODULE=on
-ENV GOPROXY=https://goproxy.cn,direct
+#ENV GOPROXY=https://goproxy.cn,direct
```
[the remark this answers is not in the document]
Comment (saitofun) [reply Dockerfile:5]: no it's necessary, if user's network was not blocked, `GOPROXY` is verb

File: README.md [hunk 7]
```diff
@@ -1,77 +1,102 @@
 # W3bstream Sprout :four_leaf_clover:
+
 W3bstream Sprout (Alpha) which supports native Halo2 circuits (WIP) as well as zkVMs like zkWASM (WIP) and RISC0.
 
-`wsctl`` is the command line tool that interact with W3bstream protocol which can be used by node operators as well as project developers.
+`wsctl`` is the command line tool that interact with W3bstream protocol which can be used by node operators as well as
+project developers.
```
[the remark this answers is not in the document]
Comment (saitofun) [reply README.md:6]: these changes in `README` caused by doc auto formatter

File: cmd/node/apis/types.go (hunk 1 above)
Comment (saitofun) [reply cmd/node/apis/types.go:17]: this type was exported, and used by other packages

File: cmd/wsctl/cmd/message_query.go (hunk 2 above)
Comment (saitofun) [reply cmd/wsctl/cmd/message_query.go:43]: defer `close()` before read/open was invoked, what's your point?

File: vm/type.go (hunk 3 above)
Comment (saitofun) [reply vm/type.go:12]: I'll fix this

File: msg/messages/message_context.go (hunk 5 above)
Comment (saitofun) [reply msg/messages/message_context.go:1]: I think `messages` is OK.

File: README.md (hunk 4 above)
Comment (saitofun) [reply README.md:50]: same above, doc auto format, I will revert it

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**